src/ui/canvas/Bitmap.hpp

```diff
@@ -106,7 +108,8 @@ class Bitmap final
 #elif defined(USE_MEMORY_CANVAS)
   WritableImageBuffer<BitmapPixelTraits> buffer = WritableImageBuffer<BitmapPixelTraits>::Empty();
 #else
-  HBITMAP bitmap = nullptr;
+  static std::unordered_map<HBITMAP, size_t> bitmap_map;
```

Comment:
Hidden global variable, and it's not thread-safe. This will crash XCSoar.

Reply:
This I don't like
What do you prefere? An use counter? I don't like the member setting to null in the source at a object copy, see line 32
Bitmap::Bitmap(Bitmap &&src)
  :bitmap(src.bitmap)
{
  src.bitmap = nullptr;
}
This means, after a copy you cannot access to the source object?

Reply:
This is not a copy constructor, it's a move constructor, and it's supposed to make the moved-from object invalid.
Why a counter?

Reply:
Unfortunately I didn't know the move constructor ;-( I learned, this is defined in C++11. With such a functional possibility their is no ref counter necessary, of course...